microprofile/telemetry/src/main/java/io/helidon/microprofile/telemetry/OpenTelemetryProducer.java

```diff
@@ -0,0 +1,193 @@
+                    }
+                    openTelemetry = OpenTelemetry.noop();
+                }
+                return openTelemetry;
+            });
+        } else {
+            if (LOGGER.isLoggable(System.Logger.Level.TRACE)) {
+                LOGGER.log(System.Logger.Level.TRACE, "Telemetry Disabled by configuration");
+            }
+            openTelemetry = LazyValue.create(OpenTelemetry::noop);
+        }
+    }
+
+    /**
+     * Get OpenTelemetry. If the OTEL Agent is present, everything is configured and prepared.
+     * We just reuse objects from the Agent.
+     *
+     * @return OpenTelemetry
+     */
+    @ApplicationScoped
+    @Produces
+    OpenTelemetry openTelemetry() {
+
+        if (HelidonOpenTelemetry.AgentDetector.isAgentPresent(config)) {
+            return GlobalOpenTelemetry.get();
+        }
+
+        return openTelemetry.get();
+    }
+
+
+    /**
+     * Provides an instance of the current OpenTelemetry Tracer.
+     *
+     * @param openTelemetry instance of OpenTelemetry.
+     * @return Tracer.
+     */
+    @Produces
+    Tracer tracer(OpenTelemetry openTelemetry) {
+        return openTelemetry.getTracer(HELIDON_SERVICE_NAME);
+    }
+
+    /**
+     * Provides an instance of the current Span.
+     *
+     * @return a {@link io.opentelemetry.api.trace.Span}.
+     */
+    @Produces
+    Span span() {
+        return Span.current();
+    }
+
+    /**
+     * Provides an instance of the current Baggage.
+     *
+     * @return a {@link io.opentelemetry.api.baggage.Baggage}.
+     */
+    @Produces
+    @ApplicationScoped
+    Baggage baggage() {
+        return Baggage.current();
+    }
+
+
+    // Process "otel." properties from microprofile config file.
+    private Map<String, String> getTelemetryProperties() {
+
+        HashMap<String, String> telemetryProperties = new HashMap<>();
+        for (String propertyName : mpConfig.getPropertyNames()) {
+            if (propertyName.startsWith("otel.")) {
+                mpConfig.getOptionalValue(propertyName, String.class).ifPresent(
+                        value -> telemetryProperties.put(propertyName, value));
+            }
+        }
+
+        // Metrics and logs exporters should be set to  "none"
+        telemetryProperties.putIfAbsent(OTEL_METRICS_EXPORTER, "none");
+        telemetryProperties.putIfAbsent(OTEL_LOGS_EXPORTER, "none");
+        return telemetryProperties;
+    }
+
+    // Check if Telemetry is disabled by checking configuration or environmental variables.
+    private boolean isTelemetryDisabled() {
+        if (telemetryProperties.get(ENV_OTEL_SDK_DISABLED) != null) {
+            return Boolean.parseBoolean(telemetryProperties.get(ENV_OTEL_SDK_DISABLED));
+        } else if (telemetryProperties.get(OTEL_SDK_DISABLED) != null) {
+            return Boolean.parseBoolean(telemetryProperties.get(OTEL_SDK_DISABLED));
+        }
+        return true;
+    }
```

Comment:
I think this could be also written like this. That way it would be shorter and easier to read from my opinion. What do you think?
```
private boolean isTelemetryDisabled() {
    return Boolean.parseBoolean(telemetryProperties.get(ENV_OTEL_SDK_DISABLED))
            || Boolean.parseBoolean(telemetryProperties.get(ENV_OTEL_SDK_DISABLED));
}
```

Reply:
Not true actually. Do not mind this comment.

Reply:
The logic is: if there is a property set: read it and return it, otherwise return true.
If both are not set, it will evaluate `false || false`, which is `false`. 
So let us keep it this way :)